draco/analysis/transform.py

```diff
@@ -158,6 +158,36 @@ def setup(self, tel):
             .reshape(-1)
         )
 
+        # Construct the equivalent reverse_map stack for the telescope instance.
+        # Note that we identify invalid products here using an index that is the
+        # size of the stack axis.
+        def pack_product_array(arr):
+
+            nfeed = arr.shape[0]
+            nprod = (nfeed * (nfeed + 1)) // 2
+
+            ret = np.zeros(nprod, dtype=arr.dtype)
+            iout = 0
+
+            for i in range(nfeed):
+                ret[iout : (iout + nfeed - i)] = arr[i, i:]
+                iout += nfeed - i
+
+            return ret
+
+        feedmask = pack_product_array(self.telescope.feedmask)
+        self.bt_rev = np.fromiter(
```

Comment:
I'm not sure about the efficiency of rebuilding from a zipped iterator here. You could reverse the order, and build the combined feedmap, feedconj array, and then do the where on top of that.